sensor/kubernetes/fake/deployment.go

```diff
@@ -411,10 +418,12 @@ func (w *WorkloadManager) manageDeployment(ctx context.Context, resources *deplo
 }
 
 func (w *WorkloadManager) manageDeploymentLifecycle(ctx context.Context, resources *deploymentResourcesToBeManaged) {
-	timer := newTimerWithJitter(resources.workload.LifecycleDuration/2 + time.Duration(rand.Int63n(int64(resources.workload.LifecycleDuration))))
-	defer timer.Stop()
+	lifecycleTimer := newTimerWithJitter(resources.workload.LifecycleDuration/2 + time.Duration(rand.Int63n(int64(resources.workload.LifecycleDuration))))
+	defer lifecycleTimer.Stop()
 
-	deploymentNextUpdate := calculateDurationWithJitter(resources.workload.UpdateInterval)
+	// Timer controlling update cadence
+	updateTimer := time.NewTimer(calculateDurationWithJitter(resources.workload.UpdateInterval))
```

Comment:
Can we reuse `newTimerWithJitter`?

```suggestion
	updateTimer := newTimerWithJitter(resources.workload.UpdateInterval)
```